### PR TITLE
feat(TD-15): Update completion status of task in backend

### PR DESCRIPTION
**Pull request by taiman-siddiqui-sazim**
**Description of change**

- Completion status of tasks can be updated in backend
- Updated completion status of tasks are fetched from the backend

**Associated Trello link**

https://trello.com/c/SROlZb9v

**Screenshot of change**

I can use the following curl command to edit a task in the backend:

```
curl -X PATCH http://localhost:3001/api/v1/tasks/61/update_completion \
-H "Content-Type: application/json" \
-d '{
  "task": {
    "completed": true
  }
}'
```

The completion status of the task is updated and fetched from the backend:

Screenshot 2024-12-19 at 6 58 59 AM

**Dev notes**

- Frontend has been updated to update and fetch completion status of tasks using this API
- Feature to delete tasks has not been implemented yet

### CODE REVIEW

File: app/controllers/api/v1/tasks_controller.rb [hunk 1]
```diff
@@ -58,7 +58,39 @@ def update
...
+        end
+
+        completed_value = params[:task][:completed]
+        unless [true, false].include?(completed_value)
+          render json: TaskSerializer.serialize_errors(
+            ['Completed must be true or false'],
+            metadata: metadata
+          ), status: :unprocessable_entity
+          return
+        end
```
Comment (ishmam-iqbal-sazim) [inline app/controllers/api/v1/tasks_controller.rb:79]: Don't think this is necessary as you're already validating in the model

File: app/controllers/api/v1/tasks_controller.rb [hunk 2]
```diff
@@ -58,7 +58,39 @@ def update
...
+      end 
+      
+      def update_completion
+        if params[:task].keys != ['completed']
+          render json: TaskSerializer.serialize_errors(
+            ['Only the completed field can be updated'],
+            metadata: metadata
+          ), status: :unprocessable_entity
+          return
+        end
```
Comment (ishmam-iqbal-sazim) [inline app/controllers/api/v1/tasks_controller.rb:70]: You should be able to use strong_parameters here as well like the other endpoints. In that case, this snippet can be removed


```
params.require(:task).permit(:completed)
```

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

File: app/controllers/api/v1/tasks_controller.rb (hunk 1 above)
Comment (taiman-siddiqui-sazim) [reply app/controllers/api/v1/tasks_controller.rb:79]: I believe there was an issue in testing where the completed field was accepting values other than _true_ or _false_ and converting them to boolean, which is why I kept this check.

**Pull request merged**